components/sts/org.wso2.carbon.identity.sts.passive.ui/src/main/java/org/wso2/carbon/identity/sts/passive/ui/PassiveSTS.java

```diff
@@ -178,14 +179,14 @@ private void sendData(HttpServletResponse httpResp, ResponseToken respToken, Str
         String htmlPage = stsRedirectPage;
         String pageWithReply = htmlPage.replace("$url", String.valueOf(respToken.getReplyTo()));
 
-        String pageWithReplyAction = pageWithReply.replace("$action", String.valueOf(action));
+        String pageWithReplyAction = pageWithReply.replace("$action", Encode.forHtml(String.valueOf(action)));
         String pageWithReplyActionResult = pageWithReplyAction.replace("$result",
-                StringEscapeUtils.escapeHtml(String.valueOf(respToken.getResults())));
+                StringEscapeUtils.escapeHtml(Encode.forHtml(String.valueOf(respToken.getResults()))));
```

Comment:
Don't use both StringEscapeUtils and Encode.forHtml. This will encode the same twice. Remove StringEscapeUtils from here